### PR TITLE
[broadlinkthermostat] Aesthetic rename and add RM Mini 

### PR DESCRIPTION
**Pull request by GiviMAD**
Signed-off-by: Miguel Álvarez <miguelwork92@gmail.com>

This is a proposal for renaming the broadlinkthermostat binding into broadlink also add a new device type "RM Mini" which is a universal controller infrared device.
Instead of adding a new binding I thought it was better to rename the current one, as the package relies in a library which supports other devices, as indicated in the readme.
Let me know in case is better to create a new binding.
To keep the autor aware @flo-02-mu.

### CODE REVIEW

**Comment by GiviMAD** [comment]
CI error seems unrelated.

**Comment by lolodomo** [comment]
> Let me know in case is better to create a new binding.

I have no definitive opinion on the two possible options.
I would be in favour of your proposal but of course this will be a breaking change for any user who previously installed this binding. At the same time, this is probably not a very used binding ?

@openhab/add-ons-maintainers : WDYT ?

**Comment by fwolter** [comment]
I think "broadlink" would be a better name, too. We discussed this partially in #11049. But I don't see much benefit of renaming this now and breaking all installations. I'd suggest to rename it with the OH 4 release (whenever that will be) and add the new device in the existing binding.

**Comment by GiviMAD** [comment]
@lolodomo, @fwolter thanks for the feedback. I will revert the package id while keeping other changes that will not break the binding.
I will put this on hold a couple of days while finishing another contribution.

File: bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java
Comment (lolodomo) [inline bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java:111]: There is no way to exit properly from learning mode ?

File: bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java
Comment (lolodomo) [inline bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java:84]: Rather a DEBUG level.

File: bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java
Comment (lolodomo) [inline bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java:91]: It looks like only ON command is supported ?
In that case, you should call handleCommand only if command is ON.

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
As mentioned in one of my review comments, you introduced a breaking change (by renaming a thing type) and I think it can be avoided, in a simple way.

File: bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java
Comment (GiviMAD) [reply bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java:102]: No, it's required due to the underling library throwing Exception direct instances.

File: bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java
Comment (GiviMAD) [reply bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java:123]: No, the method checkData from the library thows Exception, only option I see will be catching it inside the method and throw a different one.

File: bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java
Comment (GiviMAD) [reply bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java:111]: It's exited when check data method is called.
One option to implement it will be to call check data and ignore the result when and off command is send to startlearning.
Should I added it?

File: bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/BroadlinkHandlerFactory.java
Comment (lolodomo) [inline bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/BroadlinkHandlerFactory.java:81]: What for the 2 unknown thing types ?
As they are supported, you must create an handler for them. If not, the things will remain uninitialized. Maybe it is what you expect ?

File: bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java
Comment (lolodomo) [reply bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java:116]: I let you decide.
At least now you added a check to handle only ON command, so it looks now better to me.

**Comment by lolodomo** [comment]
There is now no more breaking changes that I can identify, that is a good point.

File: bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/BroadlinkHandlerFactory.java
Comment (lolodomo) [reply bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/BroadlinkHandlerFactory.java:81]: But I see no real interest to create things if they are not handled by the binding.

File: bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java
Comment (GiviMAD) [reply bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java:116]: I'll keep it like this then, I don't think it will be used.

File: bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/BroadlinkHandlerFactory.java
Comment (GiviMAD) [reply bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/BroadlinkHandlerFactory.java:81]: Yes, that is from the original implementation, it was also strange for me.
I think it just to show you there are other devices compatibles with the protocol but not supported, I think that by returning null thing remains uninitialized.
Not sure what to do, I can completely remove the unknown thing types.

Comment (lolodomo) [reply bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/BroadlinkHandlerFactory.java:81]: Yes, I think it would be better to completely remove the unknown thing types.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**
